File: server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/tracking_tool/mingle_tracking_tool_representer.rb
##########################################################################
# Copyright 2015 ThoughtWorks, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
##########################################################################

module ApiV1
  module Config
    module TrackingTool
      class MingleTrackingToolRepresenter < ApiV1::BaseRepresenter
        alias_method :mingle, :represented

        ERROR_KEYS = {
          'baseUrl'           => 'base_url',
          'projectIdentifier' => 'project_identifier'
        }

        property :base_url
        property :project_identifier
        property :mql_grouping_conditions, exec_context: :decorator

        def mql_grouping_conditions
          mingle.getMqlCriteria().getMql
        end

        def mql_grouping_conditions=(value)
          mingle.setMqlCriteria(value)
        end


      end
    end
  end
end

File: api/api-shared-v10/src/main/java/com/thoughtworks/go/apiv10/admin/shared/representers/trackingtool/MingleTrackingToolRepresenter.java
/*
 * Copyright 2019 ThoughtWorks, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thoughtworks.go.apiv10.admin.shared.representers.trackingtool;

import com.thoughtworks.go.api.base.OutputWriter;
import com.thoughtworks.go.api.representers.JsonReader;
import com.thoughtworks.go.config.MingleConfig;

public class MingleTrackingToolRepresenter {

    public static void toJSON(OutputWriter jsonWriter, MingleConfig mingleConfig) {
        jsonWriter.add("base_url", mingleConfig.getBaseUrl());
        jsonWriter.add("project_identifier", mingleConfig.getProjectIdentifier());
        jsonWriter.add("mql_grouping_conditions", mingleConfig.getMqlCriteria().getMql());
    }

    public static MingleConfig fromJSON(JsonReader jsonReader) {
        MingleConfig mingleConfig = new MingleConfig();
        jsonReader.readStringIfPresent("base_url", mingleConfig::setBaseUrl);
        jsonReader.readStringIfPresent("project_identifier", mingleConfig::setProjectIdentifier);
        jsonReader.readStringIfPresent("mql_grouping_conditions", mingleConfig::setMqlCriteria);
        return mingleConfig;
    }
}
